RNIronSourceInterstitials.js

```diff
@@ -0,0 +1,57 @@
+import { NativeModules, NativeEventEmitter } from 'react-native';
+
+const RNIronSourceInterstitials = NativeModules.RNIronSourceInterstitials;
+const IronSourceInterstialsEventEmitter = new NativeEventEmitter(RNIronSourceInterstitials);
+
+const eventHandlers = {
+  interstitialDidLoad: new Map(),
+  interstitialDidShow: new Map(),
+  interstitialDidFailToShowWithError: new Map(),
+  didClickInterstitial: new Map(),
+  interstitialDidClose: new Map(),
+  interstitialDidOpen: new Map(),
+  interstitialDidFailToLoadWithError: new Map()
+};
+
+const addEventListener = (type, handler) => {
+  switch (type) {
+    case 'interstitialDidLoad':
+    case 'interstitialDidShow':
+    case 'interstitialDidFailToShowWithError':
+    case 'didClickInterstitial':
+    case 'interstitialDidClose':
+    case 'interstitialDidOpen':
+    case 'interstitialDidFailToLoadWithError':
+      eventHandlers[type].set(handler, IronSourceInterstialsEventEmitter.addListener(type, handler));
+      break;
+    default:
+      console.log(`Event with type ${type} does not exist.`);
+  }
+};
+
+const removeEventListener = (type, handler) => {
```

Comment:
This will cause an exception if you pass in an unknown `type`.  We should probably account for that by performing a null check against the type map, just to be on the safe side, though we should definitely log an exception, or throw a specific exception stating the reason why this fails.